pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -201,28 +217,29 @@ public boolean orderNumberExists(String orderNumber) {
             SimpleResult dbVals = new SimpleResult();
             try (PreparedStatement selectKitRequest = conn.prepareStatement(SQL_SELECT_ORDER_NUMBER)) {
                 selectKitRequest.setString(1, orderNumber);
-                try (ResultSet rs = selectKitRequest.executeQuery()) {
-                    if (rs.next()) {
-                        dbVals.resultValue = true;
-                    } else {
-                        dbVals.resultValue = false;
-                    }
-
-                } catch (Exception e) {
-                    throw new RuntimeException("Error getting values from db", e);
-                }
-            } catch (SQLException ex) {
+                dbVals.resultValue = checkIfOrderNumberExists(selectKitRequest);
+            } catch (Exception ex) {
                 dbVals.resultException = ex;
             }
             return dbVals;
         });
 
         if (results.resultException != null) {
-            throw new RuntimeException("Error checking if values exist in db", results.resultException);
+            throw new DsmInternalError("Error checking if values exist in db", results.resultException);
         }
         return (boolean) results.resultValue;
     }
 
+    private Boolean checkIfOrderNumberExists(PreparedStatement selectKitRequest) throws SQLException {
```

Comment:
This change seems to lessen the readability. Was there a problem reported by the linter?